src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -1,164 +1,581 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
 using Roslyn.Utilities;
-using System.Collections.Immutable;
-using System.Collections.Generic;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed partial class LocalRewriter
     {
-        public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
+        private struct IsPatternTranslator : IDisposable
         {
-            var loweredExpression = VisitExpression(node.Expression);
-            var loweredPattern = LowerPattern(node.Pattern);
-            return MakeIsPattern(loweredPattern, loweredExpression);
-        }
+            private readonly LocalRewriter _localRewriter;
+            private readonly SyntheticBoundNodeFactory _factory;
 
-        // Input must be used no more than once in the result. If it is needed repeatedly store its value in a temp and use the temp.
-        BoundExpression MakeIsPattern(BoundPattern loweredPattern, BoundExpression loweredInput)
-        {
-            var syntax = _factory.Syntax = loweredPattern.Syntax;
-            switch (loweredPattern.Kind)
-            {
-                case BoundKind.DeclarationPattern:
-                    {
-                        var declPattern = (BoundDeclarationPattern)loweredPattern;
-                        return MakeIsDeclarationPattern(declPattern, loweredInput);
-                    }
+            private readonly ArrayBuilder<BoundExpression> _conjunctBuilder;
+            private readonly ArrayBuilder<BoundExpression> _sideEffectBuilder;
```

Comment:
>_sideEffectBuilder [](start = 59, length = 18)

Please add comments explaining what these builders are accumulating.

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec](https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec) of [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156540469](https://github.com/dotnet/roslyn/pull/23209#discussion_r156540469) [](ancestors = 156540469)